src/builtin.jq

```diff
@@ -42,6 +42,7 @@ def to_entries: [keys_unsorted[] as $k | {key: $k, value: .[$k]}];
 def from_entries: map({(.key // .Key // .name // .Name): (if has("value") then .value else .Value end)}) | add | .//={};
 def with_entries(f): to_entries | map(f) | from_entries;
 def reverse: [.[length - 1 - range(0;length)]];
+def dig($path): reduce ($path | split("."))[] as $key (.; .[$key]);
```

Comment:
I would avoid using `split/1` since I consider it basically deprecated (all the other `split` and `splits` functions take a regex, but `split/1` doesn't and it is kept like that for backward compatibility). I would especially avoid it with `"."` since in regexps, `.` matches any characters and if `split/1` aligns with the other `split*` functions in the future, this will not work anymore.

So, using string division, it would be:

```suggestion
def dig($path): reduce ($path / ".")[] as $key (.; .[$key]);
```

But also notice that this is a really silly way of writing:
```jq
def dig($path): getpath($path / ".");
```
So maybe use that instead. To be honest I don't think this function makes much sense as a builtin.